### PR TITLE
feat: add typegen for sway contracts

### LINKED ISSUE #220: (feat) Integrate typegen / fuels CLI for ts-sdk niceties
https://docs.fuel.network/docs/fuels-ts/fuels-cli/generating-types/

### PR DESCRIPTION
**Pull request by DefiCake**
Closes #220 

As side effects:
- fuel-core is updated to 0.31
- added a deploy script for the L2 bridge
- I do not know why but the CI is also acting up with some code format issue on the rust unit test, so I ran "cargo fmt + cargo clippy --fix"

### CODE REVIEW

File: packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs [hunk 1]
```diff
@@ -212,7 +212,7 @@ mod tests {
         assert!(
             matches!(error_receipt,
                 Error::Transaction(Reason::Reverted {reason, ..})
-                if reason == "NotOwner".to_string()
+                if reason == *"NotOwner"
```
Comment (K1-R1) [inline packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs:215]:
```suggestion
                if reason == "NotOwner"
```

Comment (K1-R1) [reply packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs:215]: Should just be able to do this if `reason` is either a String or &str

File: packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs [hunk 2]
```diff
@@ -255,7 +255,7 @@ mod tests {
         assert!(
             matches!(error_receipt,
                 Error::Transaction(Reason::Reverted {reason, ..})
-                if reason == "IdentityZero".to_string()
+                if reason == *"IdentityZero"
```
Comment (K1-R1) [inline packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs:258]:
```suggestion
                if reason == "IdentityZero"
```

File: packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs [hunk 3]
```diff
@@ -310,7 +310,7 @@ mod tests {
         assert!(
             matches!(error_receipt,
                 Error::Transaction(Reason::Reverted {reason, ..})
-                if reason == "NotOwner".to_string()
+                if reason == *"NotOwner"
```
Comment (K1-R1) [inline packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs:313]:
```suggestion
                if reason == "NotOwner"
```

File: packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs [hunk 4]
```diff
@@ -434,7 +434,7 @@ mod tests {
...
```
Comment (K1-R1) [inline packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs:437]:
```suggestion
                if reason == "NotOwner"
```

File: packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs [hunk 5]
```diff
@@ -478,7 +478,7 @@ mod tests {
         assert!(
             matches!(error_receipt,
                 Error::Transaction(Reason::Reverted {reason, ..})
-                if reason == "IdentityZero".to_string()
+                if reason == *"IdentityZero"
```
Comment (K1-R1) [inline packages/fungible-token/bridge-fungible-token/tests/functions/proxy/mod.rs:481]:
```suggestion
                if reason == "IdentityZero"
```

File: packages/test-utils/src/deploy-scripts/bridge.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  DeployContractResult,
+  Provider,
+  Wallet,
+  WalletUnlocked,
+  ZeroBytes32,
+} from 'fuels';
+
+const { L1_TOKEN_GATEWAY, L2_SIGNER, L2_RPC } = process.env;
+
+// This helper avoids an exception in case that the contract
```
Comment (K1-R1) [inline packages/test-utils/src/deploy-scripts/bridge.ts:24]:
```suggestion
// This helper avoids an exception in the case that the contract
```

**Review by K1-R1 — APPROVED** [review]
Some nits but otherwise LGTM

**Review by K1-R1 — APPROVED** [review]

**Review by viraj124 — APPROVED** [review]

**Pull request merged**